filament/src/details/Texture.cpp

```diff
@@ -465,7 +472,44 @@ void FTexture::generateMipmaps(FEngine& engine) const noexcept {
         return;
     }
 
-    engine.getDriverApi().generateMipmaps(mHandle);
+    DriverApi& driver = engine.getDriverApi();
+    driver.generateMipmaps(mHandle);
+    updateLodRange(driver, 0, mLevelCount);
+}
+
+bool FTexture::canHaveTextureView() const noexcept {
+    // TODO: this will eventually include swizzling
+    return any(mUsage & Usage::SAMPLEABLE) && mLevelCount > 1;
+}
+
+void FTexture::updateLodRange(DriverApi& driver, uint8_t baseLevel, uint8_t levelCount) const noexcept {
+    if (any(mUsage & Usage::SAMPLEABLE) && mLevelCount > 1) {
+        auto& range = mLodRange;
+        uint8_t const last = int8_t(baseLevel + levelCount);
+        if (range.first > baseLevel || range.last < last) {
+            range.first = std::min(range.first, baseLevel);
```

Comment:
I'm confused by the logic here. If `range` starts out as `{0,0}` (denoting all LODs available), won't `range.first` always be 0?

Reply:
yes good catch, that's a bug.